### ISSUE TITLE
Type attributes are added to javascript and CSS, this is not needed in HTML5

### ISSUE THREAD

**Issue by mattyrob** [ClassicPress/ClassicPress#339]
JavaScript and CSS resources queued in the current code includes a `type` attribute which is not necessary in HTML5. Themes should be able to direct core to not include this attribute when inserting JavaScript and CSS into the main theme. A WordPress ticked and patch exists for the same issue, it should be considered here too.

https://core.trac.wordpress.org/ticket/42804

## Expected behavior
JavaScript and CSS should be queued without the `type` attribute in HTML5 themes.

## Current behavior
JavaScript and CSS are currently queued with the `type` attribute in all circumstances unless hacked around.

## Possible solution
See ticket linked above with a proposed and patched solution.

## Steps to reproduce (for bugs)

1. Set up a vanilla install of ClassicPress
2. Validate the source on a HTML5 checker, note the warnings relating to the `type` attribute.

**Comment by mattyrob** [comment]
I've started to work on this and one issue I've run into is that a couple of the Emoji JS files as loaded via a javascript loader: ` src/wp-includes/js/wp-emoji-loader.js`

Is there an elegant way to detect if a theme supports HTML5 from within the source JavaScript files?

**Comment by mattyrob** [comment]
Initial branch created in my repository with some changes to the core code and some unit tests attempted.

https://github.com/mattyrob/ClassicPress/tree/339

PR not created yet as I'm inviting comments and feedback please.

**Comment by nylen** [comment]
I just took another look at this, and the related discussion on WP Trac, and I'm now wondering _why_ it is so important to preserve compatibility for schemes older than HTML5? This was just assumed without explanation in the Trac ticket.

Given that extensive effort was put into HTML5 in order to make sure it would work as specified with basically any legacy browser, it would be a lot simpler to just say "ClassicPress v2 sites are HTML5, you shouldn't expect them to validate under HTML4/etc".

What do you think?

In any case I think it would be worth putting this latest commit as a patch on the Trac ticket and having basically the same discussion there.

**Comment by mattyrob** [comment]
@nylen 

I guess the backwards compatible approach was the WP way - try to code around any possible breaking changes.

I have no major desire to keep compatibility and recommending an HTML5 compatible theme would be a positive move - HTML5 has been around since 2008 and recommended for 4 years officially now so it should be the first choice for themes IMO.

I've amended as many of the tests as I could find so far but dropping legacy support could simplify some of the tests I've made and also resolve the issue I was facing with the `src/wp-includes/js/wp-emoji-loader.js` file.

I'll create a PR with what I have so far to progress some discussion.
